hermes-management/src/main/java/pl/allegro/tech/hermes/management/api/ReadOnlyFilter.java

```diff
@@ -0,0 +1,69 @@
+package pl.allegro.tech.hermes.management.api;
+
+import static javax.servlet.http.HttpServletResponse.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.filter.GenericFilterBean;
+import pl.allegro.tech.hermes.management.domain.mode.ModeService;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+@WebFilter(urlPatterns = "/*")
+public class ReadOnlyFilter extends GenericFilterBean {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReadOnlyFilter.class);
+    private static final String READ_ONLY_ERROR_MESSAGE = "Action forbidden due to read-only mode";
+    private static final Set<String> whitelist = new HashSet<>();
+
+    private final ModeService modeService;
+
+    public ReadOnlyFilter(ModeService modeService) {
+        this.modeService = modeService;
+
+        initWhitelist();
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+
+        if (modeService.isReadOnlyEnabled()) {
+            if (!req.getMethod().equals("GET") && !isWhitelisted(req.getRequestURI())) {
+                HttpServletResponse resp = ((HttpServletResponse) response);
+                resp.sendError(SC_SERVICE_UNAVAILABLE, READ_ONLY_ERROR_MESSAGE);
```

Comment:
I am not sure if 503 is correct it this case. It indicates that something is wrong with server. In my opinion 403 will be better.

Reply:
I thought about that too, but!
4xx - are client errors
5xx - are server errors
Read only means that it's a server status where server not able to process something.
So 503 looks more convenient.

Reply:
I am assuming that we will handle READ_ONLY mode also in UI, and on server side we would like to validate if client behaves correctly.